crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/FullAssessmentAvailabilityService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.laa.crime.meansassessment.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.meansassessment.model.common.ApiCreateMeansAssessmentRequest;
+import uk.gov.justice.laa.crime.meansassessment.model.common.ApiCreateMeansAssessmentResponse;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.InitialAssessmentResult;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
+import static uk.gov.justice.laa.crime.meansassessment.staticdata.enums.MagCourtOutcome.COMMITTED_FOR_TRIAL;
+import static uk.gov.justice.laa.crime.meansassessment.staticdata.enums.NewWorkReason.HR;
+
+@Slf4j
+@Service
+public class FullAssessmentAvailabilityService {
+
+    public void processFullAssessmentAvailable(final ApiCreateMeansAssessmentRequest meansAssessmentRequest,
+                                               final ApiCreateMeansAssessmentResponse meansAssessmentResponse) {
+
+        log.info("Start full assessment available check for create means assessment request {} and response {}",
+                meansAssessmentRequest, meansAssessmentResponse);
+
+        requireNonNull(meansAssessmentRequest, "meansAssessmentRequest must not be null");
+        requireNonNull(meansAssessmentResponse, "meansAssessmentResponse must not be null");
+
+        meansAssessmentResponse.setFullAssessmentAvailable(false);
+
+        if (!Objects.isNull(meansAssessmentRequest.getAssessmentDate())) {
+            meansAssessmentResponse.setFullAssessmentAvailable(true);
+        } else {
+            switch (InitialAssessmentResult.valueOf(meansAssessmentResponse.getResult())) {
+                case FULL: {
+                    meansAssessmentResponse.setFullAssessmentAvailable(true);
+                    break;
+                }
+                case FAIL: {
+                    processFullAssessmentAvailableOnResultFail(meansAssessmentRequest, meansAssessmentResponse);
+                    break;
+                }
+                case HARDSHIP: {
+                    ofNullable(meansAssessmentRequest.getNewWorkReason()).ifPresent(newWorkReason -> {
```

Comment:
This is also enforced through the schema, this will never be null.

Reply:
right ok. resolved